helix_fhir_client_sdk/fhir_client.py

```diff
@@ -318,20 +320,37 @@ def get(self) -> FhirGetResponse:
                     response_json: Dict[str, Any] = json.loads(text)
                     # see if this is a Resource Bundle and un-bundle it
                     if (
-                        self._auto_unbundle
-                        and "resourceType" in response_json
+                        "resourceType" in response_json
                         and response_json["resourceType"] == "Bundle"
                     ):
                         # resources.append(text)
                         if "entry" in response_json:
-                            # iterate through the entry list
-                            # have to split these here otherwise when Spark loads them it can't handle
-                            # that items in the entry array can have different types
                             entries: List[Dict[str, Any]] = response_json["entry"]
                             entry: Dict[str, Any]
+                            resources_dict: Dict[str, Any] = {}
                             for entry in entries:
                                 if "resource" in entry:
-                                    resources.append(json.dumps(entry["resource"]))
+                                    if self._separate_bundle_resources:
+                                        # iterate through the entry list
+                                        # have to split these here otherwise when Spark loads them it can't handle
+                                        # that items in the entry array can have different types
+                                        resource_type: str = entry["resource"][
+                                            "resourceType"
+                                        ]
+                                        if (
+                                            resource_type.lower()
+                                            not in resources_dict.keys()
+                                        ):
+                                            resources_dict[resource_type.lower()] = []
+                                        # add to list
+                                        resources_dict[resource_type.lower()].append(
+                                            entry["resource"]
+                                        )
+                                    else:
+                                        resources.append(json.dumps(entry["resource"]))
+                            if self._separate_bundle_resources:
```

Comment:
should this be if NOT

Reply:
the test pass becasue`resources_dict` is the separated bundle so when `_separate_bundle_resources` is true we add it.